src/libmongoc/tests/test-mcd-nsinfo.c

```diff
@@ -0,0 +1,95 @@
+#include <mongoc/mcd-nsinfo.h>
+
+#include "TestSuite.h"
+#include "test-conveniences.h" // ASSERT_MATCH
+
+static void
+test_nsinfo_works (void)
+{
+   mcd_nsinfo_t *nsinfo = mcd_nsinfo_new ();
+
+   // Add several namespaces.
+   bson_error_t error;
+   ASSERT_OR_PRINT (0 <= mcd_nsinfo_append (nsinfo, "db.coll1", &error), error);
+   ASSERT_OR_PRINT (0 <= mcd_nsinfo_append (nsinfo, "db.coll2", &error), error);
+   ASSERT_OR_PRINT (0 <= mcd_nsinfo_append (nsinfo, "db.coll3", &error), error);
+
+   // Check resulting indexes.
```

Comment:
Confirming: are the indices expected to be consistently sequenced in the order the nsinfo were appended? If so, recommend documenting this behavior (deterministic indices). If not, recommend changing these assertions to `0 <=` instead of asserting specific indices.

Reply:
Indexes are expected to be assigned in the order they are added. I expect it is needed to pass specification tests (example) that assert the contents of the `nsInfo` sent.